test/spec/SearchResults/getFacetValues-facetOrdering.js

```diff
@@ -0,0 +1,1518 @@
+          }
+        }
+      }
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('brand', {
+      sortBy: ['name:desc'],
+      facetOrdering: true
+    });
+
+    var expected = [
+      {count: 511, isRefined: false, name: 'Samsung'},
+      {count: 386, isRefined: true, name: 'Apple'},
+      {count: 551, isRefined: false, name: 'Insignia™'}
+    ];
+
+    expect(facetValues).toEqual(expected);
+  });
+
+  test('facetOrdering: false without sortBy uses default order', function() {
+    var data = require('./getFacetValues/disjunctive.json');
+    var order = {
+      renderingContent: {
+        facetOrdering: {
+          values: {
+            brand: {
+              order: ['Samsung', 'Apple', 'Insignia™']
+            }
+          }
+        }
+      }
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('brand', {
+      facetOrdering: false
+    });
+
+    var expected = [
+      {count: 386, isRefined: true, name: 'Apple'},
+      {count: 551, isRefined: false, name: 'Insignia™'},
+      {count: 511, isRefined: false, name: 'Samsung'}
+    ];
+
+    expect(facetValues).toEqual(expected);
+  });
+
+  test('without facetOrdering, nor sortBy', function() {
+    var data = require('./getFacetValues/disjunctive.json');
+    var order = {
+      renderingContent: {}
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('brand');
+
+    var expected = [
+      {count: 386, isRefined: true, name: 'Apple'},
+      {count: 551, isRefined: false, name: 'Insignia™'},
+      {count: 511, isRefined: false, name: 'Samsung'}
+    ];
+
+    expect(facetValues).toEqual(expected);
+  });
+});
+
+describe('hierarchical facet', function() {
+  test('empty facet ordering', function() {
+    var data = require('./getRefinements/hierarchical-cards.json');
+    var order = {
+      renderingContent: {
+        facetOrdering: {}
+      }
+    };
+    var results = data.content.results.slice();
+    results[0] = Object.assign(order, results[0]);
+
+    var searchParams = new SearchParameters(data.state);
+    var result = new SearchResults(searchParams, results);
+
+    var facetValues = result.getFacetValues('hierarchicalCategories');
+
+    var expected = {
+      name: 'hierarchicalCategories',
+      count: null,
+      isRefined: true,
+      path: null,
+      exhaustive: true,
+      data: [
```

Comment:
That's a lot of values, so it's easy to loose track of the expectation. Any chance we can reduce this dataset to its minimal?

Reply:
yes good point, I'll copy the dataset and remove some